examples/packetdump.rs

```diff
@@ -172,30 +190,48 @@ fn handle_ipv6_packet(interface_name: &str, ethernet: &EthernetPacket) {
 fn handle_arp_packet(interface_name: &str, ethernet: &EthernetPacket) {
     let header = ArpPacket::new(ethernet.payload());
     if let Some(header) = header {
-        println!("[{}]: ARP packet: {}({}) > {}({}); operation: {:?}",
-                 interface_name,
-                 ethernet.get_source(),
-                 header.get_sender_proto_addr(),
-                 ethernet.get_destination(),
-                 header.get_target_proto_addr(),
-                 header.get_operation());
+        println!(
+            "[{}]: ARP packet: {}({}) > {}({}); operation: {:?}",
+            interface_name,
+            ethernet.get_source(),
+            header.get_sender_proto_addr(),
+            ethernet.get_destination(),
+            header.get_target_proto_addr(),
+            header.get_operation()
+        );
     } else {
         println!("[{}]: Malformed ARP Packet", interface_name);
     }
 }
 
-fn handle_packet(interface_name: &str, ethernet: &EthernetPacket) {
+fn handle_ethernet_packet(interface: &NetworkInterface, ethernet: &EthernetPacket) {
+    let interface_name = &interface.name[..];
     match ethernet.get_ethertype() {
         EtherTypes::Ipv4 => handle_ipv4_packet(interface_name, ethernet),
         EtherTypes::Ipv6 => handle_ipv6_packet(interface_name, ethernet),
         EtherTypes::Arp => handle_arp_packet(interface_name, ethernet),
         _ => {
-            println!("[{}]: Unknown packet: {} > {}; ethertype: {:?} length: {}",
-                     interface_name,
-                     ethernet.get_source(),
-                     ethernet.get_destination(),
-                     ethernet.get_ethertype(),
-                     ethernet.packet().len())
+            if interface.is_up() && interface.is_point_to_point() {
+                // TUN
```

Comment:
What is the `EtherType` field set to in this case, where a TUN interface is used? Just want to know what's going on here.

Reply:
`TUN` interface packet type is `IP_Packet(IPv4/IPv6)` , so they havn't `ether_type` .

in `libpnet` , `ether_type` filed must be `0` .